geode-docs/tools_modules/micrometer/micrometer-configuration.html.md.erb

```diff
@@ -0,0 +1,120 @@
+  private static final String HOSTNAME = "localhost";
+  private static final int PORT = getInteger(PORT_PROPERTY, DEFAULT_PORT);
+
+  private static Logger LOG = getLogger(SimpleMetricsPublishingService.class);
+
+  private final int port;
+  private PrometheusMeterRegistry registry;
+  private HttpServer server;
+
+  public SimpleMetricsPublishingService() {
+    this(PORT);
+  }
+
+  public SimpleMetricsPublishingService(int port) {
+    this.port = port;
+  }
+
+  @Override
+  public void start(MetricsSession session) {
+    registry = new PrometheusMeterRegistry(DEFAULT);
+    session.addSubregistry(registry);
+
+    InetSocketAddress address = new InetSocketAddress(HOSTNAME, port);
+    server = null;
+    try {
+      server = HttpServer.create(address, 0);
+    } catch (IOException thrown) {
+      LOG.error("Exception while starting " + getClass().getSimpleName(), thrown);
+    }
+    HttpContext context = server.createContext("/");
+    context.setHandler(this::requestHandler);
+    server.start();
+
+    int boundPort = server.getAddress().getPort();
+    LOG.info("Started {} http://{}:{}/", getClass().getSimpleName(), HOSTNAME, boundPort);
+  }
+
+  private void requestHandler(HttpExchange httpExchange) throws IOException {
+    final byte[] scrapeBytes = registry.scrape().getBytes();
+    httpExchange.sendResponseHeaders(200, scrapeBytes.length);
+    final OutputStream responseBody = httpExchange.getResponseBody();
+    responseBody.write(scrapeBytes);
+    responseBody.close();
+  }
+
+  @Override
+  public void stop(MetricsSession session) {
+    session.removeSubregistry(registry);
+    registry = null;
+    server.stop(0);
+  }
+}
+```
+
+
+To make your service discoverable, add the following provider-configuration file in the resource directory of your publishing service jar file:
+
+```pre
+META-INF/services/org.apache.geode.metrics.MetricsPublishingService
+```
+Add a line inside the file indicating the fully qualified class name of your implementation:
+
+```pre
+my.domain.SimpleMetricsPublishingService
+```
+Note: The above APIs are considered experimental. Micrometer metrics is a new addition to <%=vars.product_name%> and the API may change.
+
+### Add Your jar File to the classpath When You Start a Server or Locator
+To add your metrics publishing service to a server or locator, add your jar file to the classpath when you start the server or locator via `GFSH` and specify the `prometheus.metrics.port` listed in SimpleMetricsPublishingService:
+
+```pre
+gfsh>create locator --name my-locator --classpath=<path-to-my-jar-file>/my.jar --J=-Dprometheus.metrics.port=9914
+ 
+gfsh>create server --name my-server --classpath=<path-to-my-jar-file>/my.jar --J=-Dprometheus.metrics.port=9915
+```
+Alternatively, you can add your jar file to the extensions directory in your Geode installation and only specify the `prometheus.metrics.port`. Then `GFSH` will add your jar file to the classpath whenever it creates a server or locator.
+
```

Comment:
Do you think it would be appropriate to link to the Micrometer example in the geode-example repo here? That example does the same thing as what this page is describing, so if they have questions after reading this page that might be a good reference.

Reply:
Due to doc tools limitations and separate example repo, we can't reliably link to examples from a user guide. We can consider a descriptive sign-post directing the reader to an example. I suggest this is beyond the scope of the immediate PR.